File: String/LongestCommonPrefix.java
import java.util.Arrays;

public class LongestCommonPrefix {

    public String longestCommonPrefix(String[] strs) {
        // Sort the array in lexicographic order (alphabetically)
        Arrays.sort(strs);
        String strStart = strs[0];
        String strLast = strs[strs.length - 1];
        int count = 0;

        // Compare characters of the first and last strings
        for (int i = 0; i < strStart.length(); i++) {
            if (strStart.charAt(i) != strLast.charAt(i)) {
                break;
            } else {
                count++;
            }
        }

        if (count == 0) {
            return "";
        } else {
            return strStart.substring(0, count);
        }
    }
}

File: String/Easy/LongestCommonPrefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # sort the array and then compare the first and last string
        strs.sort()
        str_start = strs[0]
        str_last = strs[-1]
        count = 0
        for i in range(len(str_start)):
            if (str_start[i] != str_last[i]):
                break
            else:
                count += 1
        return "" if count==0 else str_start[0:count]